### ISSUE TITLE
feature request: Support exclude(ignore) using .riceignore 

### ISSUE THREAD

**Issue by at15** [GeertJohan/go.rice#83]
Similar issues: #80 #37

go.rice does not support excluding files and this result in huge binary if you have folders like `node_modules`, and due to the way rice find boxes (related #47 ), it's hard to specify different box location for same box during development and production. So similar to git and docker, which has `.gitignore` and `.dockeringore`, a `.riceignore` file can be introduced to make rice more flexible without
changing how rice find boxes.

The [implementation](https://github.com/docker/docker/tree/master/builder/dockerignore) in docker for `dockerignore` can be a good start, also there is a [dockerignore library](https://github.com/codeskyblue/dockerignore)

I have already tried it in my own [fork](https://github.com/at15/go.rice/tree/feature/ignore) in [this commit](https://github.com/at15/go.rice/commit/a21c2dfc0680aa0eb3db0ec200e988fef45d2f34). It works for append zip only, and use the library mentioned above, it works for [this small project](https://github.com/dyweb/Ayi/commit/25503a697fb6f849099dea782c008f517ebbc52d). Though the behavior is not exactly same as `gitingore` and only `.riceignore` file at box root is supported.

I can make a pull request if go.rice is still under development. Thanks~

**Comment by producerism** [comment]
This would be a great feature, currently having to manually edit my rice-box.go to remove DS_STORE files and others

**Comment by nadilas** [comment]
Has this been implemented?

**Comment by mdone-cloudian** [comment]
No, please, not an ignore option, or at least also provide a include-only thing. `rice` is embedding my project's `.git` directory *by default*. This is a terrible default behavior.

**Comment by GeertJohan** [comment]
@mdone-cloudian Is your .git directory in the assets box you're embedding?

**Comment by mdone-cloudian** [comment]
Please be patient, it will take me a couple of days until I can revisit this and give input. We're in release mode here at `$WORK`.

**Comment by mdone-cloudian** [comment]
[Two months later...] Ok, the way I solved it (and I guess it was the original intended method) was to put the files in its own subdir and point `rice` to it. I find all the example show this, so I guess it was more an error on my side. It's all good now, forget I ever said anything :)
